R/utilities-parameters.R

```diff
@@ -6,12 +6,20 @@
 #' @param sheets Names of the excel sheets containing the information about the
 #'   parameters. Multiple sheets can be processed. If no sheets are provided,
 #'   the first one in the Excel file is used.
+#' @param columnTypes Either `NULL` to guess all from the spreadsheet or a
+#'   character vector containing one entry per column from these options:
+#'   "skip", "guess", "logical", "numeric", "date", "text" or "list". If exactly
+#'   one `col_type` is specified, it will be recycled. The content of a cell in
+#'   a skipped column is never read and that column will not appear in the data
+#'   frame output. A list cell loads a column as a list of length 1 vectors,
+#'   which are typed using the type guessing logic from `columnTypes = NULL`, but
+#'   on a cell-by-cell basis.
 #'
 #' @return A list containing vectors 'paths' with the full paths to the
 #'   parameters, 'values' the values of the parameters, and 'units' with the
 #'   units the values are in.
 #' @export
-readParametersFromXLS <- function(paramsXLSpath, sheets = NULL) {
+readParametersFromXLS <- function(paramsXLSpath, sheets = NULL, columnTypes = NULL) {
```

Comment:
columnTypes can be hard coded here - character, character, numeric, character.

Reply:
The problem with hard-coding like this is that reading sheets like these won't work:

> Error: Sheet 3 has 11 columns, but `col_types` has length 4.

![image](https://user-images.githubusercontent.com/11330453/167625092-3692fe33-ed8e-4707-8952-7df7e3933c85.png)